src/Mod/Part/Gui/ViewProvider2DObject.h

```diff
@@ -67,14 +67,16 @@ class PartGuiExport ViewProvider2DObjectGrid : public ViewProvider2DObject
     App::PropertyEnumeration GridStyle;
     App::PropertyBool TightGrid;
     App::PropertyBool GridSnap;
-    App::PropertyBool GridAutoSize;
+    App::PropertyBool GridAuto; //Change size of grid based on view area.
+    App::PropertyBool GridAutoSize; //TODO: Find if this has any use.

```

Comment:
I did this to get feedback from reviewers. It's not described, there's a code snippet that is completely not understandable. It may be to you, but because you already know. Such properties should be described in the code.

Reply:
Properties are implicitly already described : https://github.com/FreeCAD/FreeCAD/blob/3d7b1163585117f1ca6ca11e9029fd1346913bad/src/Mod/Part/Gui/ViewProvider2DObject.cpp#L69

Moreover I can't really get your comment because any decent IDE should be able to point you where such a variable is used... :confused:

Reply:
I did find all the references of this GridAutoSize properties. (Though I'm interested to know how you make this research in VS. When I right click and do 'find all references' the results is often lacking.)

What I meant is that I can't understand what this code at line 246 does in FC and when it is called in FC.
Please explain me what it is used for in freeCad, what workbench and what circumpstances.

Reply:
I now almost nothing about VS usage. :) Probably would help to use a more Qt-aware IDE as QtCreator or KDevelop.

I don't really get your point in understanding the behavior. As described, `GridAutoSize` role is to "Autosize grid base on shape boundbox".
So on line 246 (and followings), you can just see that the bounding box of the object is computed (if shape changed / line 245), and then the grid is redrawn accordingly if enabled.

Reply:
Hmmm. This isn't a setting per se, in the sense that's it's not targeting to be changed by the user.
This is only a property that view providers can use to enable/disable the feature (what VPSketch does by disabling it).
If you want to see the effect, you can change this in the Property pane or using some Python. ;)

Reply:
Haaa so we're getting to it. ;)

Ok so it's not intended to the user, and VPSketch set it to false in ctor and doesn't touch it. So sketcher doesn't use it.
And we already mentioned that only sketcher is using grid.
So we can conclude that this is dead code.

Property pane doesn't show it when I select a sketch.

Hardcoding 'true' in VPSketch just crash FC.

My guess is that someone intended to make the grid auto-resize as I'm doing now, and just dropped it and forgot about it.

Reply:
> It's not intended to the user, and VPSketch set it to false in ctor and doesn't touch it. So sketcher doesn't use it. And we already mentioned that only sketcher is using grid. So we can conclude that this is dead code.

This is a wrong conclusion. :) This code feature can be enabled by the user playing with some properties of the sketch. This isn't easy though. ;)

> Property pane doesn't show it when I select a sketch.

Do you have "Show all" ?
> Hardcoding 'true' in VPSketch just crash FC.

Can't reproduce in master. Maybe it's due to your PR.
> My guess is that someone intended to make the grid auto-resize as I'm doing now, and just dropped it and forgot about it.

This is an actual feature --even if unused--. The grid can be resized based on bounding box dimension of the object. ;)

Reply:
Oh I see now I can activate it.

So ok it seems to be a feature. Which needs to be activated, not only in the property editor (which is already over the head of 95% of users **imo**).
But additionally you have to show hidden properties. It feels like a super useful feature :p

On top of that it doesn't even update dynamically. You have to close and re-open sketch for it to change.

Another property that sound useless is tightgrid. The wiki description isn't very clear : https://wiki.freecadweb.org/Property_editor
It sounds like something that has been developped to mitigate the enormous amount of grid lines that is due to drawing the lines centered at origins.

Reply:
> So ok it seems to be a feature. Which needs to be activated, not only in the property editor (which is already over the head of 95% of users **imo**). But additionally you have to show hidden properties. It feels like a super useful feature :p

Again, today it's not intended to me manipulated by the user. ;)
> On top of that it doesn't even update dynamically. You have to close and re-open sketch for it to change.

Yes, saw that. Some flaws that may have to be fixed. :)

> Another property that sound useless is tightgrid. The wiki description isn't very clear : https://wiki.freecadweb.org/Property_editor It sounds like something that has been developped to mitigate the enormous amount of grid lines that is due to drawing the lines centered at origins.

I don't know exactly what it does (or is supposed to do).